Task10 function

monicayun opened the pull request:
added avgDim function to RandomMonomialIdeals file
@Sondzus

monicayun committed:
copy pasted function

monicayun committed:
fixed syntax

Sondzus commented:
Great! thanks for staring to port this over to the package. Couple comments: 
1) It is good that you fixed the syntax to make variable declarations local.
2) I do not understand why "TypicalValue => RR" is not there any more? Was there an issue? The method is indeed supposed to return a real number, so it seems like that was the correct thing to have there. 
3) I hesitate to merge this immediately until we address post 25 on Piazza - since this will certainly require input changes to the method.

Commit:
Typical Value

Added Typical Value return

Commit:
TEMPORARY CHANGE

Must change later, added ideal in loop so that it changes the list to
an ideal. Need to take out later

Commit:
Merge branch 'master' into task10-function

Commit:
Merge branch 'master' into task10-function

monicayun committed:
took out N and Z from inputs

Commit:
Merge remote-tracking branch 'origin/task10-function' into task10-function

# Conflicts:
#	RandomMonomialIdeals.m2

monicayun committed:
fixed syntax

Commit:
Fixed merge errors and added N and Z

N and Z are found in the function and no longer an input. Also fixed
merge issues

Commit:
TEMP CHANGE

Commented out the file writing, so we can test the function. Must
Change later

monicayun committed:
deleted space

monicayun committed:
Merge branch 'master' into task10-function

Commit:
inverted changes

inverted some changes and added some syntax

Commit:
Edits to avgDim

Made a lot of edits so that it will run. Some are renaming, deleting :,
deleting filename and fileNameExt which we need to put in later…

Commit:
Merge branch 'master' into task10-function

monicayun committed:
made BaseFileName, FileNameExt and ShowDimensionTally optional inputs

ShowDimensionTally prints the histogram when assigned true. The default
if false for ShowDimenstionTally with empty strings for BaseFileName
and FileNameExt
If no optional inputs are given the function returns just the avgDim

monicayun committed:
Merge branch 'master' into task10-function

Commit:
Merge branch 'master' into task10-function

# Conflicts:
#	RandomMonomialIdeals.m2

monicayun committed:
deleted conflicts

exported optional inputs for avgDim and the function avgDim

Commit:
change

test

Commit:
deleted stuff

deleted a lot of unnecessary things

monicayun committed:
Merge remote-tracking branch 'origin/task10-function' into task10-function

# Conflicts:
#	RandomMonomialIdeals.m2

monicayun committed:
added function and exported terms back

Commit:
changed name

Change name from avgDim to dimStats

Commit:
edits to dimStats for zero ideals

Changed the code so that it can deal with zero ideals. comments out
unnecessary code. Changes input to want a list of ideals.

Sondzus commented:
@monicayun @parkerjoncus  Are you done with doc and tests? Once you are feel free to leave a comment and tag me (Sondzus) so I can take a look at the work and check your pull request. :)

Commit:
Merge branch 'master' into task10-function

monicayun committed:
added section for tests

monicayun committed:
added in tests and changed return values

Sondzus committed:
Merge branch 'master' into task10-function

Commit:
Added ///

# Conflicts:
#	RandomMonomialIdeals.m2

Commit:
added tally check to tests

checked to see that tally values added up to N

Commit:
documentation for ShowDimensionTally

Added the documentation for the optional input of ShowDimensionTally

monicayun committed:
added some documentation (incomplete)

# Conflicts:
#	RandomMonomialIdeals.m2

monicayun committed:
made small fix

monicayun committed:
fixed small error

monicayun committed:
fixed output

there was an issue where it would return an empty sequence if
ShowDimensionTally was false. I added some parenthesis and it seems to
be working now. I don’t think it liked the more than one thing in the
then statement

monicayun committed:
added example in documentation and deleted a bunch of stuff

deleted extra lines left over from the copy paste

Commit:
added more examples

Added more examples that are more like what a user would do. Also
changed a spelling error

Commit:
ShowDimensionTally example

added an example to ShowDimensionTally so that more than one value will
have a tally.

Commit:
edit in ShowDImTally example

Changed example inputs from ideals to listOfIdeals

Sondzus commented:
Hi @monicayun @parkerjoncus  I think there is no need to worry about the optional inputs for writing to file. Let's just have a simple function that returns the dim statistics to the user.
(Streamlined file writing will be taken care of in the new Sample object that @rosborn  will provide later.)

Commit:
edit to text and examples

Put the similar examples and text together

parkerjoncus commented:
@Sondzus alright, so you want us to take them completely out of the function or just leave it how it is and just don't document it.

Sondzus commented:
@parkerjoncus  I think just remove it. 
(If you prefer, you can just comment out all the file-writing lines for now, if you want to keep a copy in case you want to write something to file later. Currently we create files and don't tell the user we do so - so that's not so great; so I say comment out or delete those lines from the method entirely. And kill the filename options too. Sorry - I realize we were first gonna write to file, but now that I know we'll have a better option to do so, it's better not to mess with all these different ways to write to all thee different files.)

Commit:
Deletion of filename

Deleted File writing code and commented out code from use before.

parkerjoncus commented:
@Sondzus alright I took out the file writing code and the commented out code from before. I figured if we had to write to file again in the code, we still have the original version in RMIs. As far as what is left, I think we just need to go over the wording of the documentation and the examples and see if we can improve anything we said or did so that it is more specific, easy to understand, or needs to be more formal.

Commit:
spelling error

monicayun committed:
changed text

made it shorter

monicayun committed:
edits to documentation

edited wording

sonjapetrovic committed:
minor changes to documentation

monicayun committed:
made small changes to wording

monicayun committed:
Merge remote-tracking branch 'origin/task10-function' into task10-function

# Conflicts:
#	RandomMonomialIdeals.m2

Commit:
Merge branch 'master' into task10-function

monicayun committed:
added link to dim

Commit:
changed names in test

replaced randomGeneratingSet and randomGeneratingSets in tests

Commit:
edits to test and name changes
